packages/@react-native-windows/telemetry/src/telemetry.ts

```diff
@@ -4,207 +4,411 @@
-      exception: e,
-      properties: props,
-    });
   }
 
-  static shouldDisable: boolean = false;
-  static preserveMessages: boolean = false;
-}
+  static setProjectInfo(
+    info: projectUtils.AppProjectInfo | projectUtils.DependencyProjectInfo,
+  ) {
+    if (!Telemetry.client) {
+      return;
+    }
 
-function getAnonymizedPath(filepath: string): string {
-  const projectRoot = process.cwd().toLowerCase();
-  filepath = filepath.replace(/\//g, '\\');
-  const knownPathsVars = ['AppData', 'LocalAppData', 'UserProfile'];
-  if (filepath.toLowerCase().startsWith(projectRoot)) {
-    const ext = path.extname(filepath);
-    const rest = filepath.slice(projectRoot.length);
-    const nodeModules = '\\node_modules\\';
-    // this is in the project dir but not under node_modules
-    if (rest.toLowerCase().startsWith('\\windows\\')) {
-      return `[windows]\\???${ext}(${filepath.length})`;
-    } else if (rest.toLowerCase().startsWith(nodeModules)) {
-      return 'node_modules' + rest.slice(nodeModules.length - 1);
-    } else {
-      return `[project_dir]\\???${ext}(${filepath.length})`;
-    }
-  } else {
-    for (const knownPath of knownPathsVars) {
-      if (
-        process.env[knownPath] &&
-        filepath.toLowerCase().startsWith(process.env[knownPath]!.toLowerCase())
-      ) {
-        return `[${knownPath}]\\???(${filepath.length})`;
-      }
+    Telemetry.projectProp = info;
+  }
+
+  static startCommand(info: CommandStartInfo) {
+    if (!Telemetry.client) {
+      return;
+    }
+
+    if (Telemetry.commandInfo.startInfo) {
+      return;
     }
+
+    Telemetry.commandInfo.startTime = Date.now();
+    Telemetry.commandInfo.startInfo = info;
+
+    // Set common command props
+    Telemetry.client!.commonProperties.commandName = info.commandName;
   }
-  return '[path]';
-}
-/**
- * Sanitize any paths that appear between quotes (''), brackets ([]), or double quotes ("").
- * @param msg the string to sanitize
- */
-export function sanitizeMessage(msg: string): string {
-  const cpuThreadId = /^\d+(:\d+)?>/g;
-  msg = msg.replace(cpuThreadId, '');
-  const parts = msg.split(/['[\]"]/g);
-  const clean = [];
-  const pathRegEx = /([A-Za-z]:|\\)[\\/]([^<>:;,?"*\t\r\n|/\\]+[\\/])+([^<>:;,?"*\t\r\n|]+\/?)/gi;
-  for (const part of parts) {
-    if (pathRegEx.test(part)) {
-      pathRegEx.lastIndex = -1;
-      let matches: RegExpExecArray | null;
-      let noPath = '';
-      let last = 0;
-      while ((matches = pathRegEx.exec(part))) {
-        noPath +=
-          part.substr(last, matches!.index - last) +
-          getAnonymizedPath(matches[0]);
-        last = matches!.index + matches![0].length;
-      }
-      clean.push(noPath);
-    } else if (part !== '') {
-      clean.push(part);
+
+  static endCommand(info: CommandEndInfo, extraProps?: Record<string, any>) {
+    if (!Telemetry.client) {
+      return;
+    }
+
+    if (!Telemetry.commandInfo.startInfo) {
+      return;
```

Comment:
Throw on misuse?

Reply:
I didn't want to create errors just for "you're using the telemetry library wrong" since we're using telemetry to report errors.